meos/src/general/lifting.c

```diff
@@ -1551,211 +1543,438 @@ tfunc_temporal_temporal(const Temporal *temp1, const Temporal *temp2,
-      result = MEOS_FLAGS_DISCRETE_INTERP(temp2->flags) ?
-        (Temporal *) tfunc_tsequenceset_tdiscseq(
-          (TSequenceSet *) temp1, (TSequence *) temp2, lfinfo) :
-        (Temporal *) tfunc_tsequenceset_tcontseq(
-          (TSequenceSet *) temp1, (TSequence *) temp2, lfinfo);
-    else /* temp2->subtype == TSEQUENCESET */
-      result = (Temporal *) tfunc_tsequenceset_tsequenceset(
-          (TSequenceSet *) temp1, (TSequenceSet *) temp2, lfinfo);
+    const TSequence *seq = TSEQUENCESET_SEQ_N(ss, i);
+    int res = eafunc_tsequence_base(seq, value, lfinfo);
+    if (lfinfo->ever && res == 1)
+      return 1;
+    else if (! lfinfo->ever && res != 1)
+      return 0;
   }
-  return result;
+  return lfinfo->ever ? 0 : 1;
 }
 
-/*****************************************************************************
- * Functions that synchronize two temporal values and apply a Boolean function
- * in a single pass using the ever semantics, that is, it stops when a true
- * value is found.
- *****************************************************************************/
+/*****************************************************************************/
 
 /**
- * @brief Synchronize the temporal values and apply to them the function
+ * @brief Apply a lifted function to a temporal value and a base value
+ * @param[in] temp Temporal value
+ * @param[in] value Base value
+ * @param[in] lfinfo Information about the lifted function
+ */
+int
+eafunc_temporal_base(const Temporal *temp, Datum value,
+  LiftedFunctionInfo *lfinfo)
+{
+  assert(temp);
+  assert(temptype_subtype(temp->subtype));
+  switch (temp->subtype)
+  {
+    case TINSTANT:
+      return eafunc_tinstant_base((TInstant *) temp, value, lfinfo);
+    case TSEQUENCE:
+      return eafunc_tsequence_base((TSequence *) temp, value, lfinfo);
+    case TSEQUENCESET:
+      return eafunc_tsequenceset_base((TSequenceSet *) temp, value, lfinfo);
+    default:
+      meos_error(ERROR, MEOS_ERR_INTERNAL_ERROR,
+        "Unexpected temporal subtype: %d", temp->subtype);
+      return -1;
+  }
+}
+
+/*****************************************************************************/
+ 
+/**
+ * @brief Synchronize the temporal values and apply to them a lifted function
  * @param[in] inst1,inst2 Temporal values
  * @param[in] lfinfo Information about the lifted function
  * @note This function is called by other functions besides the dispatch
- * function #efunc_temporal_temporal and thus the overlapping test is
+ * function #eafunc_temporal_temporal and thus the overlapping test is
  * repeated
  */
 static int
-efunc_tinstant_tinstant(const TInstant *inst1, const TInstant *inst2,
+eafunc_tinstant_tinstant(const TInstant *inst1, const TInstant *inst2,
   LiftedFunctionInfo *lfinfo)
 {
+  assert(inst1); assert(inst2); assert(inst1->temptype == inst2->temptype);
   /* The following is ensured by the period bound test in the dispatch
    * function */
   assert(inst1->t == inst2->t);
```

Comment:
As explained in the comment before the assert, the dispatch function eafunc_temporal_temporal has a bounding box test so that when we arrive here we never need to return a -1 value.